scipy/special/tests/test_logsumexp.py

```diff
@@ -194,3 +194,26 @@ def test_softmax_multi_axes():
     x3d = x.reshape(2, 2, 2)
     assert_allclose(softmax(x3d, axis=(1, 2)), expected.reshape(2, 2, 2),
                     rtol=1e-13)
+
+
+def test_log_softmax():
```

Comment:
These tests should be split up. I would write a `TestLogSoftmax` class and have these as various methods.

Reply:
Also, there are no tests for the `axis` argument.

Reply:
I added new file test_log_softmax.py and wrote several tests in it, including one for axis argument.